scripts/merge_truth_per_tract.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+
+"""
+Merge truth catalogs for a given tract and match to object catalog
+Author: Yao-Yuan Mao
+"""
+import os
+import re
+from argparse import ArgumentParser, RawTextHelpFormatter
+
+import numpy as np
+from astropy.coordinates import SkyCoord
+import pandas as pd
+
+__all__ = ["merge_truth_per_tract", "match_object_with_merged_truth"]
+
+
+def merge_truth_per_tract(input_dir, validate=False, silent=False, **kwargs):
+    """Merge all the truth catalogs in one single tract directory
+
+    Parameters
+    ----------
+    input_dir : str
+        Path to the input tract directory
+
+    Optional Parameters
+    ----------------
+    validate : bool, optional (default: False)
+        If true, check the tract column has only one value
+    silent : bool, optional (default: False)
+        If true, turn off most printout.
+    """
+    my_print = (lambda x: None) if silent else print
+
+    my_print("Merging all files in", input_dir)
+    files_to_merge = sorted(os.listdir(input_dir))
+    df_to_merge = []
+    for filename in files_to_merge:
+        my_print("Loading", filename)
+        df = pd.read_parquet(os.path.join(input_dir, filename))
+        m = re.search(r"_hp(\d+)\b", filename)
+        if m is not None:
+            hp = int(m.groups()[0])
+            df["cosmodc2_hp"] = hp
+            df["cosmodc2_id"] = df["id"]
+            df["id"] = df["id"].astype(str)
+        else:
+            df["cosmodc2_hp"] = -1
+            df["cosmodc2_id"] = -1
+        df["is_sn"] = True if filename.startswith("sn_") else False
+        df_to_merge.append(df)
+    df = pd.concat(df_to_merge, ignore_index=True)
+    del df_to_merge
+
+    if validate:
+        assert len(np.unique(df["tract"].values)) == 1
+
+    return df
+
+
+def match_object_with_merged_truth(truth_cat, object_cat, validate=False, silent=False, **kwargs):
+    """Match object catalog with truth catalog
+
+    Parameters
+    ----------
+    truth_cat : str or pd.DataFrame
```

Comment:
Change order to
```truth_cat : pd.DataFrame or str```
and similarly for object_cat argument for consistency with descriptive line below